docs/sphinx/using/hardware.rst

```diff
@@ -220,8 +220,87 @@ The number of shots for a kernel execution can be set through
 the ``shots_count`` argument to ``cudaq.sample`` or ``cudaq.observe``. By default,
 the ``shots_count`` is set to 1000.
 
-.. code:: python 
+.. code:: python
 
     cudaq.sample(kernel, shots_count=10000)
 
 To see a complete example for using IonQ's backends, take a look at our :ref:`Python examples<python-examples-for-hardware-providers>`.
+
+
+IQM
+==================================
+
+Setting Credentials
+`````````````````````````
+
+Programmers of CUDA Quantum may access the IQM Server from either C++ or Python. Following the `quick start guide <https://iqm-finland.github.io/cortex-cli/readme.html#using-cortex-cli>`__, install `iqm-cortex-cli` and login to initialize the tokens file.
+The path to the tokens file can either be passed explicitly via an environment variable or it will be loaded automatically if located in
+the default location :code:`~/.cache/iqm-cortex-cli/tokens.json`.
+
+.. code:: bash
+
+    export IQM_TOKENS_FILE="path/to/tokens.json"
+
+Submission from C++
+`````````````````````````
+
+To target quantum kernel code for execution on an IQM Server,
+pass the ``--target iqm`` flag to the ``nvq++`` compiler, along with a specified ``--iqm-machine``.
+
+.. note::
+    The ``--iqm-machine`` is  a mandatory argument. This provided architecture must match
+    the device architecture that the program has been compiled against. The hardware architecture for a
+    specific IQM Server may be checked  via `https://<IQM server>/cocos/quantum- architecture`.
```

Comment:
It looks like there's an extra space in here. Perhaps it should be?
```suggestion
    specific IQM Server may be checked  via `https://<IQM server>/cocos/quantum-architecture`.
```

Reply:
Updated the `scripts/validate_container.sh`